resources/splunk_install.rb

```diff
@@ -0,0 +1,214 @@
+    raise "Install at #{install_dir} already exists!" if install_state['installations'][install_dir]
+    install_state['current_installation'] = install_state['installations'][install_dir] = {
+      name: name,
+      package: package,
+      version: version,
+      build: build,
+      x64: x64_support
+    }
+
+    converge_if_changed :version, :build do
+      remote_file package_path.to_s do
+        source package_url
+        show_progress true if defined? show_progress # Chef 12.9 feature
+        notifies :delete, "remote_file[#{package_path}]", :delayed
+      end
+    end
+  end
+
+  action :uninstall do
+    install_state = node.run_state['splunk_ingredient']
+    install_state.delete('current_installation')
+    install_state['installations'].delete(install_dir)
+
+    directory install_dir do
+      recursive true
+      action :delete
+    end
+  end
+end
+
+###################################
+### Platform Specific Providers ###
+###################################
+# rubocop:disable Documentation
+
+class LinuxInstall < SplunkInstall
+  resource_name :splunk_install
+  provides :splunk_install, os: 'linux'
+
+  def kernel_string
+    x64_support ? 'Linux-x86_64.tgz' : 'Linux-i686.tgz'
+  end
+
+  action :install do
+    super()
+
+    converge_if_changed :version, :build do
+      tar_extract package_path do
+        action :extract_local
+        target_dir install_dir
+      end
+    end
+  end
+end
+
+class RedhatInstall < SplunkInstall
+  resource_name :splunk_install
+  provides :splunk_install, platform_family: 'rhel'
+
+  def kernel_string
+    x64_support ? 'linux-2.6-x86_64.rpm' : 'i386.rpm'
+  end
+
+  action :install do
+    super()
+
+    converge_if_changed :version, :build do
+      rpm_package package_name do
+        source package_path.to_s
+        action :install
+      end
+    end
+  end
+
+  action :uninstall do
+    rpm_package package_name do
+      action :remove
+    end
+
+    super()
+  end
+end
+
+class DebianInstall < SplunkInstall
+  resource_name :splunk_install
+  provides :splunk_install, platform_family: 'debian'
+
+  def kernel_string
+    x64_support ? 'linux-2.6-amd64.deb' : 'linux-2.6-intel.deb'
+  end
+
+  action :install do
+    super()
+
+    converge_if_changed :version, :build do
+      dpkg_package package_name do
+        source package_path.to_s
+        action :install
+      end
+    end
+  end
+
+  action :uninstall do
+    dpkg_package package_name do
+      action :purge
+    end
+
+    super()
+  end
+end
+
+class WindowsInstall < SplunkInstall
+  resource_name :splunk_install
+  provides :splunk_install, os: 'windows'
+
+  def kernel_string
+    x64_support ? 'x64-release.msi' : 'x86-release.msi'
+  end
+
+  action :install do
+    super()
+
+    converge_if_changed :version, :build do
+      windows_package package_name do
+        source package_path.to_s
+        action :install
+        options 'LAUNCHSPLUNK=0 INSTALL_SHORTCUT=0 AGREETOLICENSE=Yes'
```

Comment:
So from the doc, install_shortcut only applies for the full splunk installation.  I'm not sure if it hurts anything to pass in additional flags but we could pretty easily flex the options based on the package attribute.